hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/table/TestHoodieTableFactory.java

```diff
@@ -165,6 +165,35 @@ void testRequiredOptionsForSource() {
 
     assertDoesNotThrow(() -> new HoodieTableFactory().createDynamicTableSource(sourceContext6));
     assertDoesNotThrow(() -> new HoodieTableFactory().createDynamicTableSink(sourceContext6));
+
+    // nested pk field is allowed
+    ResolvedSchema schema6 = SchemaBuilder.instance()
+        .field("f0",
+            DataTypes.ROW(DataTypes.FIELD("id", DataTypes.INT()), DataTypes.FIELD("date", DataTypes.VARCHAR(20))))
```

Comment:
Can we write a IT test in `ITTestHoodieDataSource`.

Reply:
Hmmm, do you mean add IT tests? 

Or remove the UT i included here and write them as IT tests?